### DISCUSSION TITLE
How to get auto generated types into the client?

### DISCUSSION THREAD

**Discussion by tresorama** [Stun3R/strapi-sdk-js#205] [Q&A]
I cannot find info about creating an instance of the strapi clien sdk that automatically return typed TS types for responses?
It is supported ?
Or we need to create types by hand ?

**Comment by Stun3R** [comment]
Do you mean the type of the content-types described in your Strapi instance or the type of the return structure (`{ attributes, meta }`)?

If you talk about the type of content-types described in your Strapi instance, we can't know what they are in the SDK, so you will have to create types by hand.

**Reply by tresorama** [reply]
I need that :
- when i create a new collection in Strapi admin the client SDK knows the schema
- so when i fecth data with the client SDK, the response include typed "items"

If i underestood right , it's this part in your response
> If you talk about the type of content-types described in your Strapi instance, we can't know what they are in the SDK, so you will have to create types by hand.

Am i right?

**Reply by Stun3R** [reply]
The Strapi admin and the client SDK don't know what they both do. So when you create a new collection in the admin, the SDK doesn't know it.
To have the correct type in the response of the fetched data, you will have to pass the collection's type you have in your frontend to the method used in the SDK:
```js
type Restaurant = {
    title: string
    // ...
}

const strapi = new Strapi({
  // Options
})

const response = await strapi.find<Restaurant[]>('restaurants')
```

**Reply by tresorama** [reply]
I've found this article from official Strapi blog. I think it can do what i need.
https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script

Have you ever used it ?
